src/services/highcharts-factory/highcharts-factory.svc.coffee

```diff
@@ -43,30 +28,33 @@ angular
             { type: 'year', count: 1, text: '1y' },
             { type: 'all', text: 'All' }
           ]
-          selected: (if _.get(xAxisOptions, 'min') then null else 0)
 
   todayUTC = moment().startOf('day').add(moment().utcOffset(), 'minutes')
 
   class Chart
-    constructor: (@id, @data = {}, @options = {})->
-      @_template = templates[@options.chartType]
+    constructor: (@id, @series = {}, @settings = {})->
+      @_template = templates[@settings.chartType]
+      @populateChartOptions(@series, @settings)
```

Comment:
Not a big fan of how this method is public, seems more like a private method, and I think everything that happens in that method could just be within the constructor block.

Reply:
Sure thing, can just move code over.